### DISCUSSION TITLE
Visualize self generated matrix on UMAP

### DISCUSSION THREAD

**Discussion by Ping-lin14** [GreenleafLab/ArchR#1481] [Questions / Documentation]
Hi everyone

I have generated a CNV matrix consisting of cell barcodes and CNV scores like this:
![image](https://user-images.githubusercontent.com/69965537/174727359-c94171f0-5790-4e45-92a5-82f6ee117b07.png)

I want to visualize it on UMAP, I tried adding the matrix using addCellColData, but it didn't work.
Does anyone have a good way?

**Comment by rcorces** [comment]
you can add one column at a time using addCellColData.
